chef_server_ha.yaml

```diff
@@ -712,7 +722,7 @@ Resources:
       TemplateURL: !Sub https://${AutomationBucket}.s3.amazonaws.com/${TemplateVersion}/chef_rds.yaml
       Parameters:
         VPC: !Sub ${VPC}
-        ChefServerSubnets: !Join [ ",", !Ref ChefServerSubnets ]
+        ChefServerSubnets: !If [UseServerSubnetsForDatabase, !Join [ ",", !Ref ChefServerSubnets ], !Join [ ",", !Ref DatabaseSubnets ] ]
```

Comment:
hey @jeffj254 I think we'd also want to follow the same convention for Elasticsearch subnets, right?

Reply:
In our particular use case, Elasticsearch will go in the same subnet as the Chef servers, but yeah I agree for completeness it should probably also be configurable in the same way as the other services.  I'll update the PR with this change

Reply:
PR updated with code for ElasticSearch